server/commands/read.go

```diff
@@ -35,25 +34,26 @@ func NewReadQuery(datastore storage.OpenFGADatastore, tracer trace.Tracer, logge
 	}
 }
 
-// Execute the ReadQuery, returning paginated `openfga.Tuple`(s) that match the tupleset
+// Execute the ReadQuery, returning paginated `openfga.Tuple`(s) that match the tuple. Return all tuples if the tuple is
+// nil or empty.
 func (q *ReadQuery) Execute(ctx context.Context, req *openfgapb.ReadRequest) (*openfgapb.ReadResponse, error) {
 	store := req.GetStoreId()
-	modelID := req.GetAuthorizationModelId()
 	tk := req.GetTupleKey()
 
+	// Restrict our reads due to some compatibility issues in one of our storage implementations.
+	if !(tk.GetObject() == "" && tk.GetRelation() == "" && tk.GetUser() == "") {
```

Comment:
The request body
```
rpc Read(ReadRequest) ...

message TupleKey {...}

message ReadRequest {
    ...
    TupleKey tuple_key = ...;
}
```
The semantics should be such that the `tuple_key` is omitted/nil if the request body is omitted. So if `tuple_key` is non-nil, then these extra validations on the non-empty tuplekey should apply.

Hence, I vote for simplification of this validation code to simply

```
if tk != nil {
    ... // old (regular) Read validation
}
```

Reply:
What if the user supplied an empty tuple
```
{
  "tuple_key": {}
}
```
then `tk` is not nil is it? It is a pointer to an empty thing `*pb.TupleKey{}`?

Reply:
As discussed offline we would like the tuple key to be `nil` in-order to read all tuples